sdk/plugin/plugin.go

```diff
@@ -78,24 +92,30 @@ func NewPluginClient(ctx context.Context, sys pluginutil.RunnerUtil, pluginRunne
 		// and version 4.
 		3: {
 			"backend": &GRPCBackendPlugin{
-				MetadataMode: isMetadataMode,
+				MetadataMode: config.IsMetadataMode,
 			},
 		},
 		4: {
 			"backend": &GRPCBackendPlugin{
-				MetadataMode: isMetadataMode,
+				MetadataMode: config.IsMetadataMode,
+			},
+		},
+		5: {
```

Comment:
We could prefer v5 if we split this to its own plugin set and run it first before falling backed to the running v3/v4 PluginSet if we had an error.

By always preferring v5 if possible we can avoid metadata mode on catalog registration as well, I think.